### PR TITLE
Add simplified callback chaining implementations

### PR DESCRIPTION
**Pull request by JaoodxD**
I've enhanced the clarity of callback function chaining in the [5-chain.js](https://github.com/HowProgrammingWorks/AsynchronousProgramming/blob/master/JavaScript/5-chain.js) example by removing certain complex concepts.

### CODE REVIEW

File: JavaScript/5a-chain.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  forward.do = (func, name) => {
+      callbacks.push([func, name]);
+      return forward;
+  };
+  
+  return forward;
+};
+
+
+// Emulate asynchronous calls
```
Comment (tshemsedinov) [inline JavaScript/5a-chain.js:31]:
```suggestion
// Emulate asynchronous calls
```

File: JavaScript/5a-chain.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+// Use list and chaining syntax to build sequence
+
+const chain = () => {
+  const callbacks = [];
+
+  const nextCallback = () => {
+    return callbacks.shift();
+  };
```
Comment (tshemsedinov) [inline JavaScript/5a-chain.js]:
```suggestion
  const nextCallback = () => callbacks.shift();
```

File: JavaScript/5a-chain.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    const [callback, name] = next;
+    console.log('Calling:', name);
+    // Pass data from previous callback
+    callback(data, forward);
+  };
+
+  forward.do = (func, name) => {
+      callbacks.push([func, name]);
+      return forward;
+  };
```
Comment (tshemsedinov) [inline JavaScript/5a-chain.js]:
```suggestion
  forward.do = (func, name) => {
    callbacks.push([func, name]);
    return forward;
  };
```

File: JavaScript/5b-chain.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+// Use list and chaining syntax to build sequence
+
+const chain = () => {
+  const callbacks = [];
+
+  const nextCallback = () => {
+    return callbacks.shift();
+  };
```
Comment (tshemsedinov) [inline JavaScript/5b-chain.js]:
```suggestion
  const nextCallback = () => callbacks.shift();
```

File: JavaScript/5b-chain.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    const [callback, args] = next;
+    console.log('Calling next with args:', { args });
+    // Pass second argument from .do
+    callback(args, forward);
+  };
+
+  forward.do = (func, name) => {
+      callbacks.push([func, name]);
+      return forward;
+  };
```
Comment (tshemsedinov) [inline JavaScript/5b-chain.js]:
```suggestion
  forward.do = (func, name) => {
    callbacks.push([func, name]);
    return forward;
  };
```

File: JavaScript/5b-chain.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+  
+  return forward;
+};
+
+
+// Emulate asynchronous calls
```
Comment (tshemsedinov) [inline JavaScript/5b-chain.js:31]:
```suggestion
// Emulate asynchronous calls
```

**Review by tshemsedinov — APPROVED** [review]
:+1:

**Comment by JaoodxD** [comment]
@tshemsedinov ty for suggestions.

**Pull request merged**